File: Daily/sbgo/Leetcode303-区域和检索.go
package sbgo

type NumArray struct {
	sum []int
}

func Constructor(nums []int) NumArray {
	sum := make([]int, len(nums)+1)
	for i, v := range nums {
		sum[i+1] = sum[i] + v
	}
	return NumArray{sum: sum}
}

func (this *NumArray) SumRange(i int, j int) int {
	return this.sum[j+1] - this.sum[i]
}

File: Daily/Rust/Leetcode303-区域和检索.rs
struct NumArray{
    sum:Vec<i32>,
}

impl NumArray {

    fn new(nums: Vec<i32>) -> Self {
        let mut sum = 0;
        let mut preSum = vec![0;nums.len() + 1];
        for i in 0..nums.len(){
            sum += nums[i];
            preSum[i + 1] = sum;
        }
        NumArray{sum:preSum}
    }
    
    fn sum_range(&self, i: i32, j: i32) -> i32 {
        self.sum[j as usize + 1] - self.sum[i as  usize]
    }
}